design/ceph-volume-provisioning.md

```diff
@@ -0,0 +1,137 @@
+      logicalDevices:
+      # bluestore: the DB, WAL, and Data are on separate LVs
+      - db: db_lv1
+        wal: wal_lv1
+        data: data_lv1
+        dbVolumeGroup: db_vg
+        walVolumeGroup: wal_vg
+        dataVolumeGroup: data_vg
+      # bluestore: the DB, WAL, and Data are all on the same LV
+      - volume: my_lv1
+        volumeGroup: my_vg
+      # filestore: data and journal on the same LV
+      - data: my_lv2
+        dataVolumeGroup: my_vg
+      # filestore: data and journal on different LVs
+      - data: data_lv3
+        dataVolumeGroup: data_vg
+        journal: journal_lv3
+        journalVolumeGroup: journal_vg
+      # devices support both filestore and bluestore configurations based on the "config.storeType" setting at the global, node, or device level
+      devices:
+      # OSD on a raw device
+      - name: sdd
+      # OSD on a partition (partition support is new)
+      - name: sdf1
+      # Multiple OSDs on a high performance device
+      - name: nvme01
+        config:
+          osdsPerDevice: 3
+```
+
+The above options for LVM and partitions look very tedious. Questions:
+- Is it useful at this level of complexity?
+- Is there a simpler way users would configure LVM?
+- Do users need all this flexibility? This looks like too many options to maintain.
+
+### Backward compatibility
+
+Rook will need to continue supporting clusters that are running different types of OSDs. All of the v0.8 OSDs must continue running
+after Rook is upgraded to v0.9 and beyond, whether they were filestore or bluestore running on directories or devices.
+
+Since `ceph-volume` only supports devices that have **not** been previously configured by Rook:
+- Rook will continue to provision OSDs directly when a `directory` is specified in the CRD
+  - Support for creating new OSDs on directories will be deprecated. While directories might still be used for test scenarios,
+  it's not a mainline scenario. With the legacy design, directories were commonly used on LVM, but LVM is now directly supported.
+  In v0.9, support for directories will remain, but documentation will encourage users to provision devices.
+- For existing devices configured by Rook, `ceph-volume` will be skipped and the OSDs will be started as previously
+- New devices will be provisioned with `ceph-volume`
+
+### Versioning
+
+Rook relies on very recent developments in `ceph-volume` that are not yet available in luminous or mimic releases.
+For example, rook needs to run the command:
+```
+ceph-volume lvm batch --prepare <devices>
+```
+
+The `batch` command and the flag `--prepare` have been added recently.
+While the latest ceph-volume changes will soon be merged to luminous and mimic, Rook needs to know if it is running an image that contains the required functionality.
```

Comment:
Add \` around `ceph-volume` as done on other occasions in the design doc (there are some more missing this).
From my perspective to keep it "unified" either have all `ceph-volume` (not commands) mentions with code tag or not. :-)